### PR TITLE
[FLINK-30593] Add tests for ScalingTracking#removeOldRecords()

### PR DESCRIPTION
**Pull request by afedulov**
This is a trivial addition of tests for removal of old records in `ScalingTracking`

### CODE REVIEW

File: flink-autoscaler/src/test/java/org/apache/flink/autoscaler/ScalingTrackingTest.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ void setUp() {
         conf.set(AutoScalerOptions.PREFER_TRACKED_RESTART_TIME, true);
     }
 
+    // ------------- getMaxRestartTimeOrDefault -------------
```
Comment (mxm) [inline flink-autoscaler/src/test/java/org/apache/flink/autoscaler/ScalingTrackingTest.java:55]: What does this comment mean?

Comment (mxm) [reply flink-autoscaler/src/test/java/org/apache/flink/autoscaler/ScalingTrackingTest.java:55]: Why is it formatted that way?

Comment (afedulov) [reply flink-autoscaler/src/test/java/org/apache/flink/autoscaler/ScalingTrackingTest.java:55]: The test was growing large and I added sections for easier navigation. The name indicates the method that is the focus for the tests in the respective section. Formatting is arbitrary. We could do something like this, in line with how code blocks are sometimes demarcated in Flink:
```
    // ------------------------------------------------------------------------
    //  Tests for getMaxRestartTimeOrDefault()
    // ------------------------------------------------------------------------
```

**Comment by afedulov** [comment]
@mxm is this the right ticket?  
[FLINK-33572] Minimize ConfigMap API operations for autoscaler state

File: flink-autoscaler/src/test/java/org/apache/flink/autoscaler/ScalingTrackingTest.java (hunk 1 above)
Comment (mxm) [reply flink-autoscaler/src/test/java/org/apache/flink/autoscaler/ScalingTrackingTest.java:55]: I would advise against these kind of comments. Similarly to JavaDocs, they tend to add more confusion than actually helping new people. Plus, they become outdated quickly.

Comment (afedulov) [reply flink-autoscaler/src/test/java/org/apache/flink/autoscaler/ScalingTrackingTest.java]: removed...

**Pull request merged**